moto/elasticbeanstalk/models.py

```diff
@@ -148,5 +159,19 @@ def _find_environment_by_arn(self, arn: str) -> FakeEnvironment:
                     return env
         raise KeyError()
 
+    def delete_application(
+        self, application_name: str, terminate_env_by_force: bool
+    ) -> FakeApplication:
+        if application_name:
+            if application_name in self.applications:
+                application = self.applications[application_name]
+                application.status = "deleting"
+                if terminate_env_by_force:
+                    application.terminate_env_by_force = terminate_env_by_force
+                else:
+                    application.terminate_env_by_force = False
+                return application
+        raise ApplicationNotFound(application_name)
```

Comment:
Would you mind adding a test for this scenario? Just to verify the error handling works correctly.

http://docs.getmoto.org/en/latest/docs/contributing/development_tips/tests.html#negative-tests

Reply:
That's right, I added the `test_delete_unknown_application` test to validate that we indeed get receive the `ApplicatioNotFound` error.

```python
@mock_elasticbeanstalk
def test_delete_unknown_application():
    conn = boto3.client("elasticbeanstalk", region_name="us-east-1")

    application_name = "myapp"
    unknown_application_name = "myapp1"

    conn.create_application(ApplicationName=application_name)
    with pytest.raises(ClientError) as exc:
        conn.delete_application(ApplicationName=unknown_application_name)
    err = exc.value.response["Error"]
    assert err["Code"] == "ApplicationNotFound"
    assert (
        err["Message"]
        == f"Elastic Beanstalk application {unknown_application_name} not found."
    )
```